qiskit_experiments/data_processing/nodes.py

```diff
@@ -30,115 +33,225 @@ def __init__(self, axis: int, validate: bool = True):
         Args:
             axis: The axis along which to average.
             validate: If set to False the DataAction will not validate its input.
+
+        Note:
+            Axis depends on data type. ``axis = 0`` indicates averaging over
+            different circuits in the experiment data. Level2 data has only this axis.
+
+                ============  =============  =====
+                `meas_level`  `meas_return`  shape
+                ============  =============  =====
+                0             `single`       np.ndarray[shots, memory_slots, memory_slot_size]
+                0             `avg`          np.ndarray[memory_slots, memory_slot_size]
+                1             `single`       np.ndarray[shots, memory_slots]
+                1             `avg`          np.ndarray[memory_slots]
+                2             `memory=True`  list
+                ============  =============  =====
```

Comment:
It is good to have this as a reminder in here. However we should emphasis that this node is not bound to these data structures. The node will simply average over whatever dimension it is given. Can you add something to this effect:
```suggestion
        Note:
            Axis depends on data type. ``axis = 0`` indicates averaging over
            different circuits in the experiment data. Level2 data has only this axis.
            As a reminder the table below shows that various data levels in Qiskit
            and their dimension. The AverageData node will simply average the
            given array over the specified axis.

                ============  =============  =====
                `meas_level`  `meas_return`  shape
                ============  =============  =====
                0             `single`       np.ndarray[shots, memory_slots, memory_slot_size]
                0             `avg`          np.ndarray[memory_slots, memory_slot_size]
                1             `single`       np.ndarray[shots, memory_slots]
                1             `avg`          np.ndarray[memory_slots]
                2             `memory=True`  list
                ============  =============  =====
```